software/last/split/main.nf

```diff
@@ -0,0 +1,36 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process LAST_SPLIT {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::last=1219" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/last:1219--h2e03b76_0"
+    } else {
+        container "quay.io/biocontainers/last:1219--h2e03b76_0"
+    }
+
+    input:
+    tuple val(meta), path(maf)
+
+    output:
+    tuple val(meta), path("*.maf.gz"), emit: maf
+    path "*.version.txt"             , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    zcat $maf | last-split $options.args | gzip --no-name > ${prefix}.split.maf.gz
```

Comment:
```suggestion
    zcat $maf | last-split $options.args | gzip --no-name > ${prefix}.maf.gz
```

Reply:
Hi @drpatelh @aleksandrabliznina the problem we try to solve with the hardcoded suffix here is that in absence of suffix, then `$maf` and `${prefix}.maf.gz` will contain the same string, and there we have some command like `cat foo | filter > foo`, which [deletes `foo` before data even has a chance to be read](https://unix.stackexchange.com/questions/558773/bending-a-pipeline-back-into-its-origin).  With your change, the module is going to always fail if no suffix is passed in the options. Is there a better way to solve our problem?  Maybe outputting `${prefix}.maf.gz` in a subfolder?